samples/extensions/sparse_image/sparse_image.cpp

```diff
@@ -0,0 +1,1532 @@
+
+	VkFence           fence;
+	VkFenceCreateInfo fence_info{};
+	fence_info.sType = VK_STRUCTURE_TYPE_FENCE_CREATE_INFO;
+	fence_info.flags = VK_FLAGS_NONE;
+
+	VK_CHECK(vkCreateFence(get_device().get_handle(), &fence_info, nullptr, &fence));
+	vkQueueBindSparse(sparse_queue, 1U, &bind_sparse_info, fence);
+	VK_CHECK(vkWaitForFences(get_device().get_handle(), 1, &fence, VK_TRUE, DEFAULT_FENCE_TIMEOUT));
+	vkDestroyFence(get_device().get_handle(), fence, nullptr);
+}
+
+/**
+ * 	@brief Return information on what mip_level the particular page_index belongs to.
+ */
+uint8_t SparseImage::get_mip_level(size_t page_index)
+{
+	uint8_t mip_level = 0xFF;
+	if (virtual_texture.mip_levels == 1U)
+	{
+		return virtual_texture.base_mip_level;
+	}
+
+	for (uint32_t i = virtual_texture.base_mip_level; i < virtual_texture.mip_levels; i++)
+	{
+		if (page_index < (virtual_texture.mip_properties[i].mip_base_page_index + virtual_texture.mip_properties[i].mip_num_pages))
+		{
+			mip_level = i;
+			break;
+		}
+	}
+	return mip_level;
+}
+
+/**
+ * 	@brief Calculate dependencies, fill the required lists and set required flags for the particular BLOCK to be updated on screen.
+ */
+void SparseImage::process_texture_block(const TextureBlock &texture_block)
+{
+	std::list<size_t> memory_index_list;
+
+	if (texture_block.old_mip_level < virtual_texture.mip_levels)        // TODO(GS): This condition seems to be completely redundant but for some reason app is crashing with vector subscript being out of range.
+	{
+		// Old value calculations and removal from the render required list
+		get_memory_dependency_for_the_block(texture_block.column, texture_block.row, texture_block.old_mip_level, memory_index_list);
+
+		while (!memory_index_list.empty())
+		{
+			size_t memory_index = memory_index_list.front();
+			memory_index_list.pop_front();
+
+			if (!virtual_texture.page_table[memory_index].fixed)
+			{
+				virtual_texture.page_table[memory_index].render_required_list.remove({texture_block.old_mip_level, texture_block.column, texture_block.row});
+			}
+		}
+	}
+
+	if (!texture_block.on_screen)
+	{
+		return;
+	}
+
+	// New value calculations and placing into update and render_required lists
+	get_memory_dependency_for_the_block(texture_block.column, texture_block.row, texture_block.new_mip_level, memory_index_list);
+
+	while (!memory_index_list.empty())
+	{
+		size_t memory_index = memory_index_list.front();
+		memory_index_list.pop_front();
+
+		virtual_texture.page_table[memory_index].render_required_list.remove({texture_block.new_mip_level, texture_block.column, texture_block.row});
+		virtual_texture.page_table[memory_index].render_required_list.push_front({texture_block.new_mip_level, texture_block.column, texture_block.row});
+
+		if (!virtual_texture.page_table[memory_index].valid)
+		{
+			virtual_texture.update_list.remove(memory_index);
+			virtual_texture.update_list.push_back(memory_index);
+		}
+		else
+		{
+			continue;
```

Comment:
This seems to be more complex than needed.
You could just move the code from lines 326-337 into the if-block and omit the else-block.